runner/src/main.rs

```diff
@@ -667,25 +655,52 @@ fn run_example(opt: &RunExamples, example: &Example) -> Step {
     // clients in the foreground.
     #[allow(clippy::collapsible_if)]
     let run_backend_server_clients: Step = if opt.run_server.unwrap_or(true) {
-        let run_server_clients = if opt.build_client.client_variant != NO_CLIENTS {
-            Step::WithBackground {
-                name: "background server".to_string(),
-                background: run_server,
-                foreground: Box::new(run_clients),
+        let run_server_clients = if example.applications.is_empty() {
+            if opt.build_client.client_variant != NO_CLIENTS {
+                run_clients
+            } else {
+                panic!("`NO_CLIENTS` variant is not supported when no applications are provided");
             }
         } else {
-            Step::Single {
-                name: "run server".to_string(),
-                command: run_server,
+            let application = example
+                .applications
+                .get(opt.application_variant.as_str())
+                .expect("Unsupported application variant");
```

Comment:
Maybe also print the application variant, and the list of available variants?

Reply:
Done